### ISSUE TITLE
High Contrast mode issues (Windows)

### ISSUE THREAD

**Issue by gtritchie** [rstudio/rstudio#12589]
### System details

    RStudio Edition : Desktop (Electron) or Server/Workbench
    RStudio Version : 2022.12.0+353
    OS Version      : Windows 10, Windows 11
    R Version       : Any

### Steps to reproduce the problem

This issue applies to RStudio Desktop on Windows (Electron), and RStudio Server/Workbench being used from a Windows web browser (Chrome/Edge).

- enable high contrast mode in Windows (<kbd>Left Alt</kbd>+<kbd>Left Shift</kbd>+<kbd>Print Screen</kbd>
    - on a full-size Mac keyboard, <kbd>F13</kbd> is the same as <kbd>Print Screen</kbd>
    - alternatively, enable via accessibility settings ("Contrast Themes" in Windows 11, "High Contrast" on Windows-10)
- start RStudio and try to use all the features
- the Electron version of RStudio does support high contrast, unlike the legacy Qt version, but there are some visual glitches; same situation with RStudio Server in Chrome/Edge on Windows
- this issue should be used to capture all such issues

### Describe the problem in detail

Can't tell when text is selected in editor or console. Opened an [issue against Ace](https://github.com/ajaxorg/ace/issues/5028). In this screenshot, I have done "Select All" but you can't tell:

![screenshot of Rstudio in high contrast mode showing editor selection](https://user-images.githubusercontent.com/10569626/213269522-14933f85-ecc9-41fb-a263-48f0726ebef4.png)

The Files Pane has outlines around each item in the list:

![screenshot of rstudio in high contrast mode showing problems in the Files pane](https://user-images.githubusercontent.com/10569626/213254353-901276ed-fdf6-4c1f-9942-61ab4040cc87.png)

No visual indicator of selected pane in the Global Options dialog; in this example, "Console" is selected:

![screenshot of rstudio in high contrast mode showing problems in global options dialog](https://user-images.githubusercontent.com/10569626/213255174-6e15faca-87a9-43cb-bc55-ec901ae1b966.png)

No visual indicator of selection in History pane; in this example one of the items is selected:

![screenshot of rstudio in high contrast mode showing History pane](https://user-images.githubusercontent.com/10569626/213257566-03c9f0f7-6dac-4f10-ab72-22e5addee840.png)

Outline button in editor toolbar invisible:

![screenshot of rstudio in high contrast mode showing invisible outline button](https://user-images.githubusercontent.com/10569626/213259177-47a3ee38-d8e5-4dbb-9159-c31d4329c7cc.png)

Environment pane has outlines around each item:

![screenshot of RStudio showing environment pane in high contrast mode](https://user-images.githubusercontent.com/10569626/213269184-d9c29f77-a27e-4073-9c39-b486bd607410.png)

### Describe the behavior you expected

- be able to use all features in high-contrast mode

### Misc

For reference, here's what the old version (Qt) looked like in high-contrast mode:

![screenshot of RStudio Qt version in high contrast mode](https://user-images.githubusercontent.com/10569626/213257019-6bc289e9-27a2-4902-b13a-f4034ce268e4.png)

- [ ] I have read the guide for [submitting good bug reports](https://github.com/rstudio/rstudio/wiki/Writing-Good-Bug-Reports).
- [ ] I have installed the latest version of RStudio, and confirmed that the issue still persists.
- [ ] If I am reporting an RStudio crash, I have included a [diagnostics report](https://support.rstudio.com/hc/en-us/articles/200321257-Running-a-Diagnostics-Report).
- [ ] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

**Comment by gtritchie** [comment]
References:
https://support.microsoft.com/en-us/windows/change-color-contrast-in-windows-fedc744c-90ac-69df-aed5-c8a90125e696
https://blogs.windows.com/msedgedev/2020/09/17/styling-for-windows-high-contrast-with-new-standards-for-forced-colors/
https://www.smashingmagazine.com/2022/03/windows-high-contrast-colors-mode-css-custom-properties/

**Comment by tannenfreund87** [comment]
This is still a problem in current versions of R-Studio:

<img width="1276" height="1389" alt="Image" src="https://github.com/user-attachments/assets/8d7ca36f-6ae3-44b8-8c53-42eb3b757feb" />

RStudio 2026.01.2+418 "Apple Blossom" Release (1d903d3bd61a40d82bf71aad8471a5a388cc1f73, 2026-03-25) for windows
Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) RStudio/2026.01.2+418 Chrome/140.0.7339.249 Electron/38.7.2 Safari/537.36, Quarto 1.8.25

While VSCode doesn't give you a broken interface.

**Comment by tannenfreund87** [comment]
Huge improvement with the current version of RStudio, text highlighting now shows the correct colours:

> RStudio 2026.05.0+218 "Golden Wattle" Release (89f6cef5d8593410108e06a82f290fd952d506c1, 2026-05-18) for windows
> Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) RStudio/2026.05.0+218 Chrome/146.0.7680.216 Electron/41.5.0 Safari/537.36, Quarto 1.9.37

<img width="1920" height="1279" alt="Image" src="https://github.com/user-attachments/assets/8b345bd0-a132-438d-8932-c974b3494f06" />

 The black borders around the items in the file pane are a stylistic choice and work quite well for distinguishing items.

<img width="742" height="609" alt="Image" src="https://github.com/user-attachments/assets/e6814444-c3a6-49da-9881-27f421a9aff2" />

The menue selection is not visible, but this an issue that affects all Electron apps, here VSCode as example:

<img width="459" height="244" alt="Image" src="https://github.com/user-attachments/assets/1e50d3ca-e6c5-4962-92cf-a708c71e6630" />

<img width="728" height="253" alt="Image" src="https://github.com/user-attachments/assets/9cf77790-cb55-442f-aa19-17e48827ca2e" />

So, this issue can be closed from my side. Many THANKS to everyone who fixed this and made RStudio more accessible!

**Comment by gtritchie** [comment]
Thanks for reporting back, glad it is working better for you. I'll close this.
